jib-gradle-plugin/CHANGELOG.md

```diff
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file.
 ### Added
 
 - Glob pattern support for `jib.extraDirectories.permissions`. ([#1200](https://github.com/GoogleContainerTools/jib/issues/1200))
+- Support for image references with both a tag and a digest. ([#1481](https://github.com/GoogleContainerTools/jib/issues/1481))
```

Comment:
I'm curious how Jib will work and respond if there's an error for a base image, e.g.,
   - tag exists but digest doesn't match
   - digest exists but tag doesn't

Will they work? If they fail, will the error message make sense and be helpful?

Reply:
I can test to be sure, but what I think happens is:

* tag exists but digest doesn't match - this will cause `of()` to fail since it does a `Preconditions` check on the format of the tag and digest, and `parse` should throw an `InvalidImageReferenceException` since the regex won't match
* digest exists but tag doesn't - this will succeed, I modified the test to loop through every combination of registry/repository/tag/digest we have instead of separating tag and digest, which means it covers null tag + existent digest.

Reply:
Sorry, I should have been clear: I am talking about referring a base image in a remote registry. For example, I say `<from><image>openjdk:8@sha256:aaaaaa... </image></from>` (i.e., the digest `sha256:aaaa...` is not the right digest of the actual `openjdk:8` image.

Reply:
I'd like to check this behavior.